### PR TITLE
Replace mentions in message content with corresponding usernames

### PR DESCRIPTION
**Pull request by Archasion**

### CODE REVIEW

**Review by Phalanxia — APPROVED** [review]
This looks good to me but I am not very familiar with Rust. Let's have @AmaranthineCodices  review it too before merging!

File: src/main.rs [hunk 1]
```diff
@@ -660,12 +662,17 @@ async fn filter_message_edit(update: &MessageUpdate, state: &State) -> Result<()
                 }
             };
 
+            let clean_message_content = crate::message::clean_mentions(
+                content,
+                update.mentions.as_ref().unwrap_or(&vec![]),
```
Comment (AmaranthineCodices) [inline src/main.rs:667]: this avoids creating a temporary `Vec` (I think it gets optimized away regardless but still)
```suggestion
                update.mentions.as_deref().unwrap_or(&[]),
```

File: src/message.rs [hunk 2]
```diff
@@ -214,7 +233,9 @@ mod test {
     use tokio::sync::RwLock;
     use twilight_model::id::Id;
 
+    use super::MentionTrait;
```
Comment (AmaranthineCodices) [inline src/message.rs:236]: prefer importing this from `twilight` itself; this makes it appear like `MentionTrait` is coming from Chrysanthemum itself
```suggestion
    use twilight_mention::Mention as MentionTrait;
```

File: src/message.rs [hunk 3]
```diff
@@ -18,6 +20,23 @@ pub(crate) struct MessageFilterFailure {
     pub(crate) context: &'static str,
 }
 
+pub(crate) fn clean_mentions(content: &str, mentions: &[Mention]) -> String {
```
Comment (AmaranthineCodices) [inline src/message.rs:23]: This has the effect of _always_ copying the content string, which is up to 4096 bytes (or more, depending on the exact details of Discord's character limit). That could be kind of expensive if the message is very long but doesn't include any mentions.

I'd like to either:
- avoid invoking this method if we have no mentions, OR
- return a `Cow<str>` or something in the case of there being no mentions at all

**Review by AmaranthineCodices — CHANGES_REQUESTED** [review]
one concern about copying the content string. otherwise minor comments but looks good

**Review by AmaranthineCodices — APPROVED** [review]

**Pull request merged**